framework/di/Container.php

```diff
@@ -523,6 +523,8 @@ public function resolveCallableDependencies(callable $callback, $params = [])
     {
         if (is_array($callback)) {
             $reflection = new \ReflectionMethod($callback[0], $callback[1]);
+        } elseif (is_object($callback)) {
```

Comment:
It's not checking if __invoke is present in an object. How about using is_callable instead?